Full Folia Support.

Yomamaeatstoes opened the pull request:
Hello!

Here is a fully ported buildable Folia supported version. I have been testing this for the last couple of weeks and it works fine.
If possible, I wouldn't mind looking into getting PremiumVanish ported as well.

HSGamer committed:
Folia Support

HSGamer committed:
shade

HSGamer committed:
fix entity target check on VisibilityChanger

Yomamaeatstoes committed:
Update README.md

SoftikGithub committed:
java.lang.UnsupportedOperationException: Must use teleportAsync while in region threading

Yomamaeatstoes committed:
Merge pull request #1 from ImNotSoftik/master

java.lang.UnsupportedOperationException: Must use teleportAsync while…

Yomamaeatstoes committed:
Update README.md

Yomamaeatstoes commented:
I did add " # Unofficial Discord https://discord.gg/aT9z7q7hX8  " in the readme, it is just for base support if any bugs are found for folia support. Feel free to remove that.

HSGamer committed:
switch to MineLib

HSGamer committed:
update maven plugins

HSGamer committed:
update MineLib

HSGamer committed:
update relocations

Yomamaeatstoes committed:
Merge branch 'LeonMangler:master' into master

ewof commented:
```
[21:27:47 WARN]: [SuperVanish] Unknown exception occurred!
[21:27:47 WARN]: [SuperVanish] Please report this issue!
[21:27:47 WARN]: [SuperVanish] Message: 
[21:27:47 WARN]: [SuperVanish]   Accessing entity state off owning region's thread
[21:27:47 WARN]: [SuperVanish] General information: 
[21:27:47 WARN]: [SuperVanish]   ServerVersion: git-Folia-"25ee657" (MC: 1.20.4)
[21:27:47 WARN]: [SuperVanish]   PluginVersion: 6.2.18
[21:27:47 WARN]: [SuperVanish]   ServerPlugins: LuckPerms v5.4.117, MvndiCore v1.0.5, Vault v1.7.3-b0-folia, PlaceholderAPI v2.11.5-DEV-null, CoreProtect v22.2, MvndiStats v1.0.3, Towny v0.100.2.5, ChestShop v3.12.3-SNAPSHOT (build 401), MvndiEquipment v1.1.14, MvndiPlayer v1.0.2, Brewery v3.1.1, squaremap v1.2.3, TownyChat v0.115, MvndiMMO v1.0, ProtocolLib v5.2.0, MvndiMobs v0.10.7, LiteBans v2.15.0, Chunky v1.3.92, SiegeWar v2.10.0, MvndiSeasons v0.3.3, MvndiGUI v${project.version}, MvndiCombat v1.1.12, BetterRTP v3.6.12, MvndiHunters v0.1, MinecraftSSH v1.1.0, ChatControlRed v10.23.3, iConomy v5.25, ChunkyBorder v1.1.50, TownyMenus v0.0.8-SNAPSHOT, MvndiTowny v1.0.0, PaperMessages v1.0, TownyBlockGlitch v1.0, MvndiCrafting v1.0.4, FlagsH v4.0.0, BigDoors vAlpha 0.1.8.50 (b-1), FarmControl v1.2.5, HexNicks v3.1.1, TAB v4.1.2, TownyWaypoints v1.2, BetterHud vbeta-17, MvndiBanHooks v1.0, BukkitGreentext v3.1.5, BetterGUI v9.6, MvndiHolders v1.2.0, InvSeePlusPlus v0.25.2-SNAPSHOT, MvndiChisel v0.4.0, MvndiSiege v1.0, spark v1.10.60, SmoothTimber v1.25.0, FreedomChat v1.5.2, MvndiGUIShop v${project.version}, GrimAC v2.3.62, InventoryRollbackPlus v1.6.14, HeadDrop v4.35, VoidWorldGenerator v1.1.0, MvndiSounds v1.0.4, MapTowny v2.2.0,
[21:27:47 WARN]: [SuperVanish] Settings:
[21:27:47 WARN]: [SuperVanish]   MsgsVersion: 6.2.18
[21:27:47 WARN]: [SuperVanish]   Settings: ||InvisibilityFeatures.NightVisionEffect>true||InvisibilityFeatures.DisableDamage>true||InvisibilityFeatures.DisableHunger>true||InvisibilityFeatures.DisableMobTarget>true||InvisibilityFeatures.OpenChestsSilently>true||InvisibilityFeatures.DisablePressurePlates>true||InvisibilityFeatures.DisablePush>false||InvisibilityFeatures.DefaultPickUpItemsOption>false||InvisibilityFeatures.ModifyTablistPackets>true||InvisibilityFeatures.ModifyTabCompletePackets>true||InvisibilityFeatures.PreventSculkSensorActivation>true||InvisibilityFeatures.PreventTurtleEggBreaking>true||InvisibilityFeatures.DisableDripLeaf>true||InvisibilityFeatures.PreventRaidTriggering>true||InvisibilityFeatures.PreventMobSpawning>true||InvisibilityFeatures.Fly.Enable>true||InvisibilityFeatures.Fly.DisableOnReappear>true||VanishStateFeatures.AutoVanishOnJoin>false||VanishStateFeatures.ReappearOnWorldChange>false||VanishStateFeatures.ReappearOnQuit>false||VanishStateFeatures.CheckPermissionOnWorldChange>false||VanishStateFeatures.CheckPermissionOnQuit>false||VanishStateFeatures.CheckPermissionOnLogin>false||IndicationFeatures.LayeredPermissions.EnableSeePermission>true||IndicationFeatures.LayeredPermissions.LayeredSeeAndUsePermissions>false||IndicationFeatures.LayeredPermissions.MaxLevel>100||IndicationFeatures.MarkVanishedPlayersAsSpectators>true||MessageOptions.FakeJoinQuitMessages.BroadcastFakeQuitOnVanish>true||MessageOptions.FakeJoinQuitMessages.BroadcastFakeJoinOnReappear>true||MessageOptions.FakeJoinQuitMessages.AnnounceVanishReappearToAdmins>true||MessageOptions.FakeJoinQuitMessages.SendMessageOnlyToAdmins>false||MessageOptions.FakeJoinQuitMessages.SendMessageOnlyToUsers>false||MessageOptions.HideRealJoinQuitMessages>true||MessageOptions.HideAdvancementMessages>true||MessageOptions.ReappearOnQuitHideLeaveMsg>true||MessageOptions.AnnounceRealJoinQuitToAdmins>true||MessageOptions.AnnounceDeathToAdmins>true||MessageOptions.RemindVanishedOnJoin>true||MessageOptions.DisplayActionBar>true||RestrictiveOptions.PreventBlockBreaking>false||RestrictiveOptions.PreventBlockPlacing>false||RestrictiveOptions.PreventHittingEntities>false||ExternalInvisibility.ServerList.AdjustAmountOfOnlinePlayers>true||ExternalInvisibility.ServerList.AdjustListOfLoggedInPlayers>true||HookOptions.EnableEssentialsHook>true||HookOptions.EnableDynmapHook>true||HookOptions.DynmapSendJoinLeaveMessages>true||HookOptions.EnableTrailGUIHook>true||HookOptions.EnablePlaceholderAPIHook>true||HookOptions.EnableMVdWPlaceholderAPIHook>true||HookOptions.EnableCitizensHook>true||CompatibilityOptions.PlayerJoinEventPriority>HIGH||CompatibilityOptions.PlayerQuitEventPriority>HIGH||MiscellaneousOptions.UpdateChecker.Enable>true||MiscellaneousOptions.UpdateChecker.NotifyAdmins>true||ConfigVersion>6.2.18||
[21:27:47 WARN]: [SuperVanish] StackTrace: 
[21:27:47 WARN]: java.lang.IllegalStateException: Accessing entity state off owning region's thread
[21:27:47 WARN]:        at io.papermc.paper.util.TickThread.ensureTickThread(TickThread.java:79)
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.entity.CraftCow.getHandle(CraftCow.java:21)
[21:28:45 INFO]: [Bannedom of Prussia|Bannedsberg] Archteus: From now to the last 3 days of the Beta Server
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.entity.CraftCow.getHandle(CraftCow.java:6)
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.entity.CraftEntity.isInWorld(CraftEntity.java:940)
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.entity.CraftEntity.isValid(CraftEntity.java:394)
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.CraftRegionAccessor.lambda$getEntities$1(CraftRegionAccessor.java:312)
[21:27:47 WARN]:        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.CraftRegionAccessor.getEntities(CraftRegionAccessor.java:308)
[21:27:47 WARN]:        at SuperVanish-6.2.18.jar//de.myzelyam.supervanish.visibility.VisibilityChanger.hidePlayer(VisibilityChanger.java:98)
[21:27:47 WARN]:        at SuperVanish-6.2.18.jar//de.myzelyam.supervanish.visibility.VisibilityChanger.hidePlayer(VisibilityChanger.java:44)
[21:27:47 WARN]:        at SuperVanish-6.2.18.jar//de.myzelyam.supervanish.commands.SubCommand.hidePlayer(SubCommand.java:36)
[21:27:47 WARN]:        at SuperVanish-6.2.18.jar//de.myzelyam.supervanish.commands.subcommands.VanishSelf.execute(VanishSelf.java:33)
[21:27:47 WARN]:        at SuperVanish-6.2.18.jar//de.myzelyam.supervanish.commands.SubCommandMgr.execute(SubCommandMgr.java:39)
[21:27:47 WARN]:        at SuperVanish-6.2.18.jar//de.myzelyam.supervanish.commands.VanishCommand.execute(VanishCommand.java:29)
[21:27:47 WARN]:        at SuperVanish-6.2.18.jar//de.myzelyam.supervanish.SuperVanish.onCommand(SuperVanish.java:169)
[21:27:47 WARN]:        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[21:27:47 WARN]:        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.CraftServer.dispatchCommand(CraftServer.java:1042)
[21:27:47 WARN]:        at org.bukkit.craftbukkit.v1_20_R3.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64)
[21:27:47 WARN]:        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73)
[21:27:47 WARN]:        at net.minecraft.commands.execution.tasks.ExecuteCommand.a(ExecuteCommand.java:32)
[21:27:47 WARN]:        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:19)
[21:27:47 WARN]:        at net.minecraft.commands.execution.UnboundEntryAction.a(UnboundEntryAction.java:8)
[21:27:47 WARN]:        at net.minecraft.commands.execution.CommandQueueEntry.a(CommandQueueEntry.java:5)
[21:27:47 WARN]:        at net.minecraft.commands.execution.ExecutionContext.a(ExecutionContext.java:101)
[21:27:47 WARN]:        at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:434)
[21:27:47 WARN]:        at net.minecraft.commands.CommandDispatcher.performCommand(CommandDispatcher.java:336)
[21:27:47 WARN]:        at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:323)
[21:27:47 WARN]:        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2285)
[21:27:47 WARN]:        at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$14(PlayerConnection.java:2245)
[21:27:47 WARN]:        at io.papermc.paper.threadedregions.EntityScheduler.executeTick(EntityScheduler.java:173)
[21:27:47 WARN]:        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1646)
[21:27:47 WARN]:        at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407)
[21:27:47 WARN]:        at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:404)
[21:27:47 WARN]:        at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:525)
[21:27:47 WARN]:        at java.base/java.lang.Thread.run(Thread.java:1583)
[21:27:47 WARN]: [SuperVanish] Please include this information
[21:27:47 WARN]: [SuperVanish] if you report the issue.
```

Yomamaeatstoes commented:
Hello, @ewof you can fix this error by disable "InvisibilityFeatures.DisableMobTarget"

Yomamaeatstoes committed:
Merge branch 'LeonMangler:master' into master

Yomamaeatstoes committed:
Merge branch 'LeonMangler:master' into master

ewof commented:
i get
```
[05:12:58 WARN]: [SuperVanish] Global task for SuperVanish v6.2.20 generated an exception
java.lang.IllegalStateException: Cannot add effects to entities asynchronously
        at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:59) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at net.minecraft.world.entity.LivingEntity.addEffect(LivingEntity.java:1167) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at net.minecraft.world.entity.LivingEntity.addEffect(LivingEntity.java:1162) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at net.minecraft.world.entity.LivingEntity.addEffect(LivingEntity.java:1153) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at org.bukkit.craftbukkit.entity.CraftLivingEntity.addPotionEffect(CraftLivingEntity.java:536) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at org.bukkit.craftbukkit.entity.CraftLivingEntity.addPotionEffect(CraftLivingEntity.java:530) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.features.NightVision.sendAddPotionEffect(NightVision.java:121) ~[SuperVanish-6.2.20.jar:?]
        at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.features.NightVision.run(NightVision.java:116) ~[SuperVanish-6.2.20.jar:?]
        at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.minelib.scheduler.common.scheduler.Scheduler.lambda$runTimer$0(Scheduler.java:64) ~[SuperVanish-6.2.20.jar:?]
        at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.minelib.scheduler.common.util.task.FoliaTask.lambda$wrapRunnable$0(FoliaTask.java:33) ~[SuperVanish-6.2.20.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler$GlobalScheduledTask.run(FoliaGlobalRegionScheduler.java:179) ~[folia-1.21.1.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.tick(FoliaGlobalRegionScheduler.java:37) ~[folia-1.21.1.jar:?]
        at io.papermc.paper.threadedregions.RegionizedServer.globalTick(RegionizedServer.java:293) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at io.papermc.paper.threadedregions.RegionizedServer$GlobalTickTickHandle.tickRegion(RegionizedServer.java:148) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:406) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:540) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
```
DisableMobTarget is off too

Seristic committed:
Fix: Ensure entity interactions run on the entity's region thread in Folia

Seristic committed:
Fix: Properly apply night vision effect in Folia

Seristic committed:
Updated NightVision for better Folia compatibility

Seristic committed:
build: restore MineLib scheduler dependencies to compile scope

Seristic committed:
chore(build): restore MineLib deps to compile; ProtocolLib 5.2.0; repositories cleanup

Yomamaeatstoes committed:
Merge pull request #4 from Seristic/fix/folia-entity-thread

Fix: Ensure entity interactions run on the entity's region thread in Folia

Yomamaeatstoes committed:
Update README.md

HSGamer committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	src/main/java/de/myzelyam/supervanish/features/SilentOpenChest.java
#	src/main/java/de/myzelyam/supervanish/visibility/hiders/PreventionHider.java

HSGamer committed:
build

HSGamer committed:
build

HSGamer committed:
fix protocollib (why am I not surprised?)

HSGamer committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	pom.xml
#	src/main/java/de/myzelyam/supervanish/SuperVanish.java
#	src/main/java/de/myzelyam/supervanish/net/UpdateNotifier.java
#	src/main/java/de/myzelyam/supervanish/visibility/VisibilityChanger.java

HSGamer committed:
remove protocol exception catch

HSGamer committed:
fix MVdWPlaceholderAPI dependency

Yomamaeatstoes committed:
Async changes

Yomamaeatstoes commented:
> [05:12:58 WARN]: [SuperVanish] Global task for SuperVanish v6.2.20 generated an exception
> java.lang.IllegalStateException: Cannot add effects to entities asynchronously
>         at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:59) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at net.minecraft.world.entity.LivingEntity.addEffect(LivingEntity.java:1167) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at net.minecraft.world.entity.LivingEntity.addEffect(LivingEntity.java:1162) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at net.minecraft.world.entity.LivingEntity.addEffect(LivingEntity.java:1153) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at org.bukkit.craftbukkit.entity.CraftLivingEntity.addPotionEffect(CraftLivingEntity.java:536) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at org.bukkit.craftbukkit.entity.CraftLivingEntity.addPotionEffect(CraftLivingEntity.java:530) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.features.NightVision.sendAddPotionEffect(NightVision.java:121) ~[SuperVanish-6.2.20.jar:?]
>         at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.features.NightVision.run(NightVision.java:116) ~[SuperVanish-6.2.20.jar:?]
>         at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.minelib.scheduler.common.scheduler.Scheduler.lambda$runTimer$0(Scheduler.java:64) ~[SuperVanish-6.2.20.jar:?]
>         at SuperVanish-6.2.20.jar/de.myzelyam.supervanish.minelib.scheduler.common.util.task.FoliaTask.lambda$wrapRunnable$0(FoliaTask.java:33) ~[SuperVanish-6.2.20.jar:?]
>         at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler$GlobalScheduledTask.run(FoliaGlobalRegionScheduler.java:179) ~[folia-1.21.1.jar:?]
>         at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.tick(FoliaGlobalRegionScheduler.java:37) ~[folia-1.21.1.jar:?]
>         at io.papermc.paper.threadedregions.RegionizedServer.globalTick(RegionizedServer.java:293) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at io.papermc.paper.threadedregions.RegionizedServer$GlobalTickTickHandle.tickRegion(RegionizedServer.java:148) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:406) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:540) ~[folia-1.21.1.jar:1.21.1-DEV-9203a7a]
>         at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]

Addressed, both of the issue. Pleas test the latest jar

Yomamaeatstoes committed:
Remove night vision spam log

Yomamaeatstoes committed:
Fix silent quit visibility check

Yomamaeatstoes committed:
Modernize Folia support

Yomamaeatstoes committed:
Merge pull request #11 from Inquisitors-transfers/master

Modernize code

Yomamaeatstoes committed:
Revise README for plugin status and contributors

Updated README to reflect changes in plugin development and removed outdated sections.

Yomamaeatstoes committed:
Update README.md

Yomamaeatstoes committed:
Update README.md

Yomamaeatstoes committed:
Update README.md
